### ISSUE TITLE
Redisgraph returning negative indices with --compact

### ISSUE THREAD

**Issue by jeeger** [RedisGraph/RedisGraph#1643]
We're running into a bug with redisgraph on the following query pattern:

```
"graph.QUERY" "DevUnitTest" "CYPHER name=\"s\" CREATE (s:scope{name:$name}) RETURN s" "--COMPACT"
"graph.QUERY" "DevUnitTest" "CYPHER ename=\"ex01\" sid=0 MATCH (s:scope) -[:contains]->(e:expr) WHERE id(s)=$sid AND e.name=$ename RETURN e" "--COMPACT"
"graph.QUERY" "DevUnitTest" "CYPHER name=\"ex01\" term=\"V1\" version=\"1\" status=\"IN_PROGRESS\" CREATE (e:expr{name:$name,status:$status,version:$version,term:$term}) RETURN e" "--COMPACT"
"graph.QUERY" "DevUnitTest" "CYPHER chldId=1 prntId=0 MATCH (prnt), (chld) WHERE (id(prnt)=$prntId AND id(chld)=$chldId) CREATE (prnt) -[e:contains]->(chld) RETURN e" "--COMPACT"
"graph.QUERY" "DevUnitTest" "CYPHER ename=\"ex01\" sid=0 MATCH (s:scope) -[:contains]->(e:expr) WHERE id(s)=$sid AND e.name=$ename RETURN e"
"graph.QUERY" "DevUnitTest" "CYPHER ename=\"ex01\" sid=0 MATCH (s:scope) -[:contains]->(e:expr) WHERE id(s)=$sid AND e.name=$ename RETURN e" "--COMPACT"
```
The output of the last query returns a negative label index for "expr", while the  non-compact query returns the correct result (see below). Queries have been executed on a docker `redislabs/redisgraph:2.4.2` container.

Without `--compact`:
```
1) 1) "e"
2) 1) 1) 1) 1) "id"
            2) (integer) 1
         2) 1) "labels"
            2) 1) "expr"
         3) 1) "properties"
            2) 1) 1) "name"
                  2) "ex01"
               2) 1) "status"
                  2) "IN_PROGRESS"
               3) 1) "version"
                  2) "1"
               4) 1) "term"
                  2) "V1"
3) 1) "Cached execution: 1"
   2) "Query internal execution time: 0.177049 milliseconds"
```

With `--compact`:

```
1) 1) 1) (integer) 1
      2) "e"
2) 1) 1) 1) (integer) 8
         2) 1) (integer) 1
            2) 1) (integer) -2
            3) 1) 1) (integer) 0
                  2) (integer) 2
                  3) "ex01"
               2) 1) (integer) 1
                  2) (integer) 2
                  3) "IN_PROGRESS"
               3) 1) (integer) 2
                  2) (integer) 2
                  3) "1"
               4) 1) (integer) 3
                  2) (integer) 2
                  3) "V1"
3) 1) "Cached execution: 1"
   2) "Query internal execution time: 0.161892 milliseconds"
```

The output of "call db.labels()":

```
1) 1) "label"
2) 1) 1) "scope"
   2) 1) "expr"
3) 1) "Cached execution: 0"
   2) "Query internal execution time: 0.195138 milliseconds"
```

**Comment by jeffreylovitz** [comment]
Hi @jeeger,

Thanks for the report and the clear steps to reproduce!

I've opened a PR that resolves this bug, the explanation for the problem can be found there - https://github.com/RedisGraph/RedisGraph/pull/1644 .

Thank you!

**Comment by swilly22** [comment]
Resolved #1644
